src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncement.java

```diff
@@ -183,6 +193,8 @@ public void announce(@NonNull Track track) {
 
         // We don't care about the utterance id. It is supplied here to force onUtteranceCompleted to be called.
         tts.speak(announcement, TextToSpeech.QUEUE_FLUSH, null, "not used");
+
+        startTrackPoint = contentProviderUtils.getLastValidTrackPoint(track.getId());
```

Comment:
startTrackPoint is the TrackPoint from the previous announcement?

What about passing it into `update()` and then pass it to `announce()`?
TrackRecordingService should have it available.

More problems with getLastValidTrackPoint: 
* what is a valid trackpoint? (it was a location previously, right?)
* it does not take care of segments
  (might be a problem if one resumes a recording, has a time-based announcement, and no new trackpoints on first announcement)

Reply:
> What about passing it into `update()` and then pass it to `announce()`? TrackRecordingService should have it available.


That was my first thought, but I decided not to use it for looking for the easier approach. I back to my first though (I thought it was a worse approach :D).


> 
>     * it does not take care of segments
>       (might be a problem if one resumes a recording, has a time-based announcement, and no new trackpoints on first announcement)


IMHO interval is a segment between two points: A and B, and every interval shows the time (and stats)   takes the user to travel from A to B. So, in that approach pauses, stops... don't take account.

The more you pause/stop, the more it takes you to go from A to B.

I was thinking on that when I implemented intervals at the begining.